clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java

```diff
@@ -769,9 +774,17 @@ private void initializeIsolatedIngestionServer() {
     storeIngestionService.addIngestionNotifier(new IsolatedIngestionNotifier(this));
     ingestionBackend = new DefaultIngestionBackend(storageMetadataService, storeIngestionService, storageService);
 
-    LOGGER.info(
-        "Starting report client with target application port: {}",
-        configLoader.getVeniceServerConfig().getIngestionApplicationPort());
+    if (serverConfig.isLeakedResourceCleanupEnabled()) {
+      LeakedResourceCleaner leakedResourceCleaner = new LeakedResourceCleaner(
+          storageService.getStorageEngineRepository(),
+          serverConfig.getLeakedResourceCleanUpIntervalInMS(),
+          storeRepository,
+          storeIngestionService,
+          storageService,
+          metricsRepository);
+      leakedResourceCleaner.start();
```

Comment:
We will also need to shut it down properly

Reply:
And you will unfortunately have to play with unit test coverage, which I don't know what to add to... you might want to game the system for this case.